### PR TITLE
Bug #244

### PR DESCRIPTION
**Pull request by pakdel**
Supporting cmd:// URLs

Additionally, replacing {TITLE}, {USERNAME} and {PASSWORD} place holders with their values from the entry.

### CODE REVIEW

File: src/gui/DatabaseWidget.cpp [hunk 1]
```diff
@@ -452,9 +453,21 @@ void DatabaseWidget::openUrl()
 
 void DatabaseWidget::openUrlForEntry(Entry* entry)
 {
-    if (!entry->url().isEmpty()) {
-        QDesktopServices::openUrl(entry->url());
+    QString UrlString = entry->url();
+    if (UrlString.isEmpty())
+        return;
+
+    UrlString.replace("{TITLE}", entry->title(), Qt::CaseInsensitive);
```
Comment (debfx) [inline src/gui/DatabaseWidget.cpp]: - Please create atomic commits, i.e. one change per commit.
- There is already Entry::resolvePlaceholders()

File: src/gui/DatabaseWidget.cpp [hunk 2]
```diff
@@ -452,9 +453,21 @@ void DatabaseWidget::openUrl()
...
+    UrlString.replace("{USERNAME}", entry->username(), Qt::CaseInsensitive);
+    UrlString.replace("{PASSWORD}", entry->password(), Qt::CaseInsensitive);
+
+    if (UrlString.startsWith("cmd://") && UrlString.length()>6){
```
Comment (debfx) [inline src/gui/DatabaseWidget.cpp]: QDesktopServices::openUrl() will be called if UrlString=="cmd://".
Better to remove the return statement and put openUrl() into an else branch.

File: src/gui/DatabaseWidget.cpp [hunk 3]
```diff
@@ -452,9 +453,21 @@ void DatabaseWidget::openUrl()
...
+            QProcess::startDetached(UrlString.right(UrlString.length()-6));
```
Comment (debfx) [inline src/gui/DatabaseWidget.cpp]: UrlString.mid(6) would be better

File: src/gui/DatabaseWidget.cpp (hunk 2 above)
Comment (pakdel) [inline src/gui/DatabaseWidget.cpp]: QString("cmd://").length() == 6, it is not QString("cmd://").length() > 6
However, a single character is still a ridiculously short command.

**Comment by debfx** [comment]
Merged. I committed two follow-up changes:
bcc3108c3d660e54fd14eb527ea4fd090ebf1564 and af84261eb61836943ae4b37ba894b239132445ae

**Pull request merged**